common/src/main/java/org/opensearch/ml/common/transport/memorycontainer/memory/MLUpdateMemoryRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.memorycontainer.memory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+public class MLUpdateMemoryRequest extends ActionRequest {
+    @Setter
+    private MLUpdateMemoryInput mlUpdateMemoryInput;
+    private String memoryContainerId;
+    private String memoryId;
+
+    @Builder
+    public MLUpdateMemoryRequest(String memoryContainerId, String memoryId, MLUpdateMemoryInput mlUpdateMemoryInput) {
+        this.memoryContainerId = memoryContainerId;
+        this.memoryId = memoryId;
+        this.mlUpdateMemoryInput = mlUpdateMemoryInput;
+    }
+
+    public MLUpdateMemoryRequest(StreamInput in) throws IOException {
+        super(in);
+        this.memoryContainerId = in.readString();
+        this.memoryId = in.readString();
+        this.mlUpdateMemoryInput = new MLUpdateMemoryInput(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(memoryContainerId);
+        out.writeString(memoryId);
+        mlUpdateMemoryInput.writeTo(out);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
```

Comment:
can we follow this: https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/model/MLUpdateModelRequest.java#L50-L58